src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -190,11 +190,20 @@ private void createConsumer(RoutingContext routingContext) {
             httpBridgeContext.getHttpSinkEndpoints().remove(sink.name());
         });
 
-        sink.open();
+        try {
+            sink.open();
 
-        sink.handle(new HttpEndpoint(routingContext), consumerId -> {
-            httpBridgeContext.getHttpSinkEndpoints().put(consumerId.toString(), sink);
-        });
+            sink.handle(new HttpEndpoint(routingContext), consumerId -> {
+                httpBridgeContext.getHttpSinkEndpoints().put(consumerId.toString(), sink);
+            });
+        } catch (Exception ex) {
+            HttpBridgeError error = new HttpBridgeError(
+                    HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                    ex.getMessage()
+            );
+            HttpUtils.sendResponse(routingContext.response(), HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                    BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+        }
```

Comment:
If the `sink.handle()` throws then do we need to clean up side-effects of the `sink.open()`? IOW can the exception result in any kind of resource leakage?

Reply:
I took a look and there should not be any resource leakage. I was thinking about the endpoint saved and not removed from the collection so having a pending reference to a not used object (not "garbeageable" then) but it's not the case.